File: Algebra_DivideFraction.py
#!/usr/bin/python
from DivideFraction import DivFraction

##
 # Dividing fractions
 ##
numerators   = [16, 9, 640, 7]
denominators = [9, 20, 27, 20]
fractions = {'numerators':numerators, 'denominators':denominators}

print('\n    Solving:\n')
# Print as fraction
for numerator in fractions['numerators']: print('{:13d}'.format(numerator), end='')
print('{}{:>12}'.format('\n', ' '), end='')
for wasted in range(len(numerators)-1): print('{}'.format('-     /      '), end='')
print('{:>1}'.format('-'))
for denominator in fractions['denominators']: print('{:13d}'.format(denominator), end='') 
print('\n\n')

# use the DivideFraction class
div_fract = DivFraction(fractions)
fraction = div_fract.doDivide()

print('{:25d}'.format(fraction['numerator']))
print('{:>25}'.format('Answer   =   -'))
print('{:25d}'.format(fraction['denominator']))


print('\n\n')

File: Algebra_DivideFraction.pl
#!/usr/bin/perl;
use strict;
use warnings;
use DIVIDEFRACTION;

# Useful variables
my (@numerators, @denominators);
my (%fraction, %fractions);

##
 # Dividing fractions
 ##
@numerators	  = (16, 9, 640, 7);
@denominators = (9, 20, 27, 20);
%fractions = (
	numerators	 => \@numerators,
	denominators => \@denominators
);

print "\n    Solving:\n";
# Print as fraction
printf("%13u", $_) for @numerators;
printf("\n%11s", " ");
print "-      /     " for 1 .. $#numerators;
printf("%2s\n", "-");
printf("%13u", $_) for @denominators;
print "\n";

# use the DivideFraction class
my $div_fract = DIVIDEFRACTION->new(\%fractions);
%fraction = %{$div_fract->doDivide()};

printf("\n%25u\n", $fraction{numerator});
printf("%25s\n", "Answer =         -");
printf("%25u\n", $fraction{denominator});


print "\n\n";